Lib/test/test_support.py

```diff
@@ -161,6 +163,23 @@ def test_temp_dir__existing_dir__quiet_true(self):
                                         f'temporary directory {path!r}: '),
                         warn)
 
+    @unittest.skipUnless(hasattr(os, "fork"), "test requires os.fork")
+    def test_temp_dir__forked_child(self):
+        """Test that a forked child process does not remove the directory."""
+        # Run the test as an external script, because it uses fork.
+        script_helper.assert_python_ok("-c", textwrap.dedent("""
+            import os
+            from test import support
+            with support.temp_cwd() as temp_path:
+                pid = os.fork()
+                if pid != 0:
+                    # parent process
+                    os.waitpid(pid, 0)  # wait for the child to terminate
+                    # make sure that temp_path is still present
+                    if not os.path.isdir(temp_path):
+                        raise AssertionError("Child removed temp_path.")
+        """))
```

Comment:
Please add a comment explaining that the child exits without removing the temporary directory.